docs/api.adoc

```diff
@@ -7,58 +7,188 @@ v2.0.0, 2016-12
 
 Detailed description of api docs here!
 
-Hosts
+Search
 -----
-
 [nested]
-* List all
+* Search for host or group.
+Specify type as  a group, host, or any to limit the search.
 +
 [source]
 ....
-  /api/host/list
+GET /api/search?type=any&name=web_hosts
 ....
 ** Example results:
 +
 [source,json]
 ----
 {
+  "status": "success",
+  "code": 200,
+  "data": {
+  "groups": [
+    "webhosts1",
+    "webhosts2"
+    ],
   "hosts": [
-    "host1",
-    "host2",
-    "host9",
-    "host10"
+   "webhost1"
+   "webhost2"
   ]
 }
 ----
 
-* Search for specific host
+Hostvars
+--------
+
+[nested]
+* Retrieve a host and it's vars.
 +
 [source]
 ....
-/api/host/search?key=hostname
+GET /api/hostvars?name=host1
 ....
 ** Example results:
 +
 [source,json]
 ----
 {
-  "hosts": [
-    "host1",
-    "host10"
-  ]
+  "status": "successful",
+  "code": 200,
+  "data": {
+    "parent_group": "group1",
+    "host": "host1",
```

Comment:
a host can not contain hosts. In this response "host" is the name of the the host that we are querying for.